metals/src/main/scala/scala/meta/internal/metals/codeactions/ConvertCommentCodeAction.scala

```diff
@@ -0,0 +1,147 @@
+        Trees.defaultTokenizerDialect(content).tokenize
+      )
+      .flatMap(_.toOption)
+  }
+
+  private def isSingleLineComment(
+      contentLines: Vector[String],
+      couldBeScalaCli: Boolean,
+  )(
+      t: Token
+  ): Boolean = t match {
+    case tc: Token.Comment =>
+      val currentLine = contentLines(t.pos.startLine)
+      val isLikelyScalaCliDirective =
+        couldBeScalaCli && isScalaCliUsingDirectiveComment(currentLine)
+      val tokenIsSingleLine = tc.pos.startLine == tc.pos.endLine
+      val tokenStartsWithDoubleSlash =
+        currentLine.slice(tc.pos.startColumn, tc.pos.startColumn + 2) == "//"
+
+      !isLikelyScalaCliDirective &&
+      tokenIsSingleLine &&
+      tokenStartsWithDoubleSlash
+    case _ => false
+  }
+
+  private def createActionIfPossible(
+      path: AbsolutePath,
+      tokens: Tokens,
+      range: l.Range,
+      isSingleLineComment: Token => Boolean,
+  ): Option[l.CodeAction] = {
+    val indexOfLineTokenUnderCursor = tokens.lastIndexWhere(t =>
+      t.pos.startLine == range.getStart.getLine
+        && t.pos.endLine == t.pos.startLine
+        && t.pos.startColumn < range.getStart.getCharacter
+        && isSingleLineComment(t)
+    )
+    if (indexOfLineTokenUnderCursor != -1) {
+      // tokens that are strictly before cursor, i.e. they end before cursor position
+      val tokensBeforeCursor = tokens.take(indexOfLineTokenUnderCursor)
+      // token under the cursor + following tokens
+      val tokensAfterCursor = tokens.drop(indexOfLineTokenUnderCursor)
+      val textEdit = createTextEdit(
+        tokensBeforeCursor,
+        tokensAfterCursor,
+        range,
+      )
+      Some(
+        CodeActionBuilder.build(
+          title = ConvertCommentCodeAction.Title,
+          kind = this.kind,
+          changes = List(path -> List(textEdit)),
+        )
+      )
+    } else {
+      None
+    }
+  }
+
+  private def createTextEdit(
+      tokensBeforeCursor: Tokens,
+      tokensAfterCursor: Tokens,
+      range: l.Range,
+  ) = {
+    val commentBeforeCursor = collectContinuousComments(
+      tokens = tokensBeforeCursor.reverse
+    ).reverse
+
+    val commentAfterCursor = collectContinuousComments(
+      tokens = tokensAfterCursor
+    )
+    val commentStart = commentBeforeCursor.headOption
+      .map(_.pos.toLsp.getStart)
+      .getOrElse(tokensAfterCursor.head.pos.toLsp.getStart)
+    val commentEnd = commentAfterCursor.lastOption
+      .map(_.pos.toLsp.getEnd())
+      .getOrElse(range.getEnd())
+
+    val commentTokens = commentBeforeCursor ++ commentAfterCursor
+    val replaceText =
+      commentTokens
+        .map(_.value.trim()) // TODO replace with strip once we drop jdk 8
+        .mkString("/* ", "\n * ", " */")
+    val pos = new l.Range(commentStart, commentEnd)
+    new l.TextEdit(pos, replaceText)
+  }
+
+  private def collectContinuousComments(
+      tokens: Seq[Token]
+  ) = {
+    tokens
+      .takeWhile {
+        case _: Token.Trivia => true
```

Comment:
You should also check here that it's not a scala-cli using directive.
In case of:
```demo.scala
//> using scala 3.3.1
//@@my class
class A
```

Reply:
Good catch, I missed that